stock_cycle_count/models/stock_warehouse.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from datetime import datetime, timedelta
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class StockWarehouse(models.Model):
+    _inherit = 'stock.warehouse'
+
+    cycle_count_rule_ids = fields.Many2many(
+        comodel_name='stock.cycle.count.rule',
+        relation='warehouse_cycle_count_rule_rel',
+        column1='warehouse_id',
+        column2='rule_id',
+        string='Cycle Count Rules')
+    cycle_count_planning_horizon = fields.Integer(
+        string='Cycle Count Planning Horizon (in days)',
+        help='Cycle Count planning horizon in days. Only the counts inside '
+             'the horizon will be created.')
+    counts_for_accuracy_qty = fields.Integer(
+        string='Inventories for location accuracy calculation',
+        default=1,
+        help='Number of latest inventories used to calculate location '
+             'accuracy')
+
+    @api.one
+    def get_horizon_date(self):
+        date = datetime.today()
+        delta = timedelta(self.cycle_count_planning_horizon)
+        date_horizon = date + delta
+        return date_horizon
+
+    @api.model
+    def _get_cycle_count_locations_search_domain(
+            self, parent):
+        domain = [('parent_left', '>=', parent.parent_left),
+                  ('parent_right', '<=', parent.parent_right),
+                  ('cycle_count_disabled', '=', False)]
+        return domain
+
+    @api.model
+    def _search_cycle_count_locations(self, rule):
+        locations = self.env['stock.location']
+        if rule.apply_in == 'warehouse':
+            locations = self.env['stock.location'].search(
+                self._get_cycle_count_locations_search_domain(
+                    self.view_location_id))
+        elif rule.apply_in == 'location':
+            for loc in rule.location_ids:
+                locations += self.env['stock.location'].search(
+                    self._get_cycle_count_locations_search_domain(loc))
+        return locations
+
+    @api.model
+    def _cycle_count_rules_to_compute(self):
+        rules = self.cycle_count_rule_ids.search([
+            ('rule_type', '!=', 'zero'), ('warehouse_ids', '=', self.id)])
```

Comment:
I think this works. But quite weird. I would have done `('warehouse_ids', 'in', [self.id])`